src/distribution_function.py

```diff
@@ -0,0 +1,239 @@
+from os import makedirs
+import numpy as np
+import plotly.graph_objects as go
+from sympy import Symbol, pi, E, sqrt, integrate, oo
+from abc import ABC, abstractmethod
+from scipy import stats
+import math
+from config import FIGS_ROOT
+
+
+class ProbabilisticDistributionHandler(ABC):
+
+    @abstractmethod
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def show_figure(self):
+        pass
+
+    @abstractmethod
+    def save_figure(self):
+        pass
+
+
+class Visualizer(object):
+
+    @staticmethod
+    def visualize(x, y, title, fig_processing) -> None:
+        '''Show or save figure.
+
+        Args:
+            x: np.ndarray
+                data points
+            y: np.ndarray
+                value of probability
+            title: str
+                figure title
+            fig_processing: str
+                show or save
+        '''
+        fig = go.Figure()
+        fig.add_trace(go.Scatter(x=x, y=y, name=title))
+        fig.update_xaxes(title='x')
+        fig.update_yaxes(title='y')
+        fig.update_layout(title=title)
+        if fig_processing == 'show':
+            fig.show()
+        elif fig_processing == 'save':
+            makedirs(FIGS_ROOT, exist_ok=True)
+            fig_path = f'{FIGS_ROOT}/{title}.png'
+            fig.write_image(fig_path)
+            return fig_path
+        else:
+            raise NotImplementedError()
```

Comment:
I added raise NotImplementedError(), so I did't change elif fig_processing == 'save'.

Reply:
If you use raise statement, write an error message. I said this before.

Reply:
Then disable the fig_processing flag and split this function into two functions. One of them is to show and return None, and the other is to save and return str, respectively. Also, write a function for the common process (line 42 ~ line 46) in this class, and both functions should call it.